### DISCUSSION TITLE
Can the dhcp server provision the ipxe script file separately from the ipxeundionly file ? Custom script with dhcp without custom recompile ?

### DISCUSSION THREAD

**Discussion by shodanx2** [ipxe/ipxe#734] [Q&A]
All my experiments so far have been with netboot.xyz
And it does work with the netboot.xyz but I want a script I can custom modify for my purpose, boot my own, locally hosted images

I can't remember where, but somewhere in the documentation that "you will need to recompile to make full use of ipxe"

I find that kind of strange, why does the script need to be compiled in ? 

Surely that will make editing the scripts very cumbersome unless I also maintain a ready to go compilation environment

**Comment by NiKiZe** [comment]
To have https support it needs a modification from stock config.

But in terms of changing script, see https://ipxe.org/howto/chainloading#breaking_the_infinite_loop

Another common approach is to have a minimal bootstrap script that chains to your own server with all the rest.

Another alternative in efi mode is to use autoexec.ipxe file.
